### PR TITLE
build: extend script/update-frontend-canister.sh

### PR DESCRIPTION
**Pull request by smallstepman**
adds a help message and two flags:
```
   -d, --development-build    build canister using cargo
   -r, --release-build        build canister using linux/amd64 docker image
   -h, --help                 print this help message
```

When working on macOS ARM, building frontend canister using linux/amd64 docker image takes a while. This script adds an option to build the frontend canister the old way - using Rust and ic-wasm that are locally installed, this speeds up the build time significantly (useful when working on e2e tests)

### CODE REVIEW

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]
Why is src/distributed/assetstoreage.wasm.gz changed in this PR?

File: scripts/update-frontend-canister.sh [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+help()
+{
+   # display help
+   echo "Build frontend canister wasm. Copies the wasm build artifact"
+   echo "and corespontind candid file (src/canisters/frontend/ic-certified-assets/assets.did)"
+   echo "to src/distributed/assetstorage.did."
+   echo
+   echo "Options:"
+   echo "  -d, --development-build    build canister using cargo"
+   echo "  -r, --release-build        build canister using linux/amd64 docker image"
```
Comment (ericswanson-dfinity) [inline scripts/update-frontend-canister.sh:13]: Please update CONTRIBUTING.md and .github/workflows/build-frontend-canister.yml with the release build option

File: .github/workflows/build-frontend-canister.yml [hunk 2]
```diff
@@ -39,6 +39,6 @@ jobs:
         run: |
           if [ "$(git diff src/distributed/assetstorage.wasm.gz)" != "" ]
           then
-            echo "src/distributed/assetstorage.wasm.gz needs to be updated; run ./scripts/update-frontend-canister.sh"
+            echo "src/distributed/assetstorage.wasm.gz needs to be updated; run ./scripts/update-frontend-canister.sh --release_build"
```
Comment (ericswanson-dfinity) [inline .github/workflows/build-frontend-canister.yml:42]:
```suggestion
            echo "src/distributed/assetstorage.wasm.gz needs to be updated; run ./scripts/update-frontend-canister.sh --release-build"
```

File: scripts/update-frontend-canister.sh [hunk 3]
```diff
@@ -12,9 +26,48 @@ fi
...
-fi
+case ${1---help} in
+
+  --development-build | -d)
+    cargo --version >/dev/null || die "Must have cargo installed."
+    ic-wasm --version >/dev/null || die "Must have ic-wasm installed."
+    if [ -z "${CARGO_HOME}" ]
+    then
+    export CARGO_HOME="${HOME}/.cargo"
+    fi
```
Comment (ericswanson-dfinity) [inline scripts/update-frontend-canister.sh:37]:
```
$ ./scripts/update-frontend-canister.sh --development-build
./scripts/update-frontend-canister.sh: line 34: CARGO_HOME: unbound variable
```

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]

File: scripts/update-frontend-canister.sh (hunk 3 above)
Comment (smallstepman) [reply scripts/update-frontend-canister.sh:37]: 🙄

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**